olp-cpp-sdk-authentication/include/olp/authentication/TokenProvider.h

```diff
@@ -25,22 +25,51 @@
 #include <utility>
 
 #include <olp/authentication/AuthenticationCredentials.h>
-#include <olp/authentication/AutoRefreshingToken.h>
+#include <olp/authentication/ErrorResponse.h>
 #include <olp/authentication/Settings.h>
-#include <olp/authentication/TokenEndpoint.h>
-#include <olp/authentication/TokenResult.h>
 #include <olp/authentication/Types.h>
 #include <olp/core/client/CancellationContext.h>
 #include <olp/core/client/OauthToken.h>
 #include <olp/core/http/HttpStatusCode.h>
-#include <olp/core/utils/WarningWorkarounds.h>
 
 namespace olp {
 namespace authentication {
 
-// Needed to avoid endless warnings from TokenRequest/TokenResult
-PORTING_PUSH_WARNINGS()
-PORTING_CLANG_GCC_DISABLE_WARNING("-Wdeprecated-declarations")
+static constexpr auto kDefaultMinimumValidity = 300ll;
+static constexpr auto kDefaultMinimumValiditySeconds =
+    std::chrono::seconds(kDefaultMinimumValidity);
+static constexpr auto kForceRefresh = std::chrono::seconds(0);
+
+template <uint64_t MinimumValidity>
+class TokenProvider;
```

Comment:
Needed?

Reply:
yes, cause I use friend for TokenProvider from another namespace(internal)

Reply:
Why do you need a friend class? I think from our point of view we marked that this class is internal, and it is not bound to the API stability and users should not use it directly, so every method can be public. If any user thinks he requires using this class alone, it is his choice.

Reply:
ok)

Reply:
ok